src/neomaril_codex/datasources.py

```diff
@@ -0,0 +1,468 @@
+        """
+        datasources = self.list_datasets(origin=origin)
+        
+        for datasource in datasources:
+            if dataset_hash == datasource.get('Id'):
+                return NeomarilDataset(
+                    dataset_hash=datasource.get('Id'),
+                    dataset_name=datasource.get('Name'),
+                    datasource_name=self.datasource_name,
+                    group=self.group,
+                    login=self.credentials[0],
+                    password=self.credentials[1],
+                    url=self.base_url
+                )
+        raise DatasetNotFoundError('Dataset hash not found!')
+
+class NeomarilDataset(BaseNeomaril):
+    """
+    Class to do operaterations in a dataset.
+
+    Attributes
+    ----------
+    login : str
+        Login for authenticating with the client.
+        You can also use the env variable NEOMARIL_USER to set this
+    password : str
+        Password for authenticating with the client.
+        You can also use the env variable NEOMARIL_PASSWORD to set this
+    url : str
+        URL to Neomaril Server.
+        Default value is https://neomaril.staging.datarisk.net,
+        use it to test your deployment first before changing to production.
+        You can also use the env variable NEOMARIL_URL to set this
+    dataset_hash : str
+        The hash that identify the dataset
+    dataset_name : str
+        The dataset defined name
+    datasource_name : str
+        Name given to the datasource, should be not null, case sensitive, have between 1 and 64 characters,
+        that could be alphanumeric including accentuation (for example: 'é', à', 'ç','ñ') and space;
+    group : str
+        Name of the group where the datasource will be inserted.
+    """
+    def __init__(self, *, dataset_hash : str, dataset_name : str, datasource_name : str, group : str, login : str, password : str, url : str) -> None:
+        super().__init__(login, password, url)
+        self.group = group
+        self.dataset_hash = dataset_hash
+        self.dataset_name = dataset_name
+        self.datasource_name = datasource_name
+
+    def get_status(self):
+        """
+        Get dataset status.
+        
+        Returns
+        ----------
+        dict
+            {
+                status : 'Succeeded',
+                log : ''
+            }
+
+        Example
+        ----------
+        >>> dataset.get_status()
+        """
+        url = f'{self.base_url}/datasets/status/{self.group}/{self.dataset_hash}'
+
+        token = refresh_token(*self.credentials, self.base_url)
+
+        response = requests.get(
+            url=url,
+            headers={'Authorization': 'Bearer ' + token},
+            timeout=60
+        )
+
+        if response.status_code == 200:
+            status = response.json().get('Status')
+            log = response.json().get('Log')
+
+            return {'status' : status, 'log' : log }
+
+    def delete(self):
```

Comment:
Shouldn't we return a message or add a log if some error occurs? Maybe adding a message when `response.status_code != 200` ...

Reply:
I assume that has the same behavior that happens with dataset. So it is solved!